grails-app/controllers/org/pih/warehouse/core/LocationController.groovy

```diff
@@ -95,6 +95,12 @@ class LocationController {
                 params.supportedActivities = supportedActivities?.empty ? [ActivityCode.NONE.id] : supportedActivities
             }
 
+            // when we are passing address with id which is empty string or null
+            // then grails try to find an address with this id instead of creating a new one
+            if (!params.address.id) {
+                params.address.remove("id")
```

Comment:
This is fine, but you could also make the inclusion of the parameter conditional.
```
<g:if test="${locationInstance?.address}">
    <g:hiddenField name="address.id" value="${locationInstance?.address?.id}"/>
</g:if>
```
I suspect adding this check to the controller is better because it hypothetically defends against more cases. But I really don't want to be doing stuff like this in all of our controllers.

So we should add a tech debt ticket to enable convertEmptyStringsToNull in application.yml.
```
grails.databinding.convertEmptyStringsToNull = true
```

Reply:
I tried to assign `null` to the address.id manually and it wasn't bound also, so I am unsure if changing this property will help, but I will take a look and let you know. I think I will go with this approach with the condition in the controller.